primitives/comb.fil

```diff
@@ -177,4 +178,42 @@ extern "comb.sv" {
    ) -> (
       out: ['G, 'L] OUT_WIDTH
    ) where 'L > 'G, IN_WIDTH > 0, OUT_WIDTH > 0;
+}
+
+// Reduces a bundle of N wires of size W to a single wire of size W*N
+// {010, 101, 111} -> 010101111
+comp ConcatBundle[W, N]<'G: 1>(
+   in[N]: ['G, 'G + 1] W
+) -> (
+   out: ['G, 'G + 1] W*N
+) where N > 0, W > 0 {
+   if N == 1 {
+      out = in{0};
+   } else {
+      // recursively reduce
+      red := new ConcatBundle[W, N-1]<'G>(in{1..N});
+      c := new Concat[W, W*(N-1)]<'G>(in{0}, red.out);
+      out = c.out;
+   }
+}
+
+// Splits a wire of size W*N into a bundle of N wires of size W
+// 010101111 -> {010, 101, 111}
+comp SplitWire[W, N]<'G: 1>(
```

Comment:
Heck yes! This was exactly what I was thinking about during our conversation last week.